lib_eio/process.mli

```diff
@@ -171,7 +174,7 @@ module Pi : sig
     val pipe :
       t ->
       sw:Switch.t ->
-      [Flow.source_ty | Resource.close_ty] r * [Flow.sink_ty | Resource.close_ty] r
+      [< Flow.source_ty | Resource.close_ty] r * [< Flow.sink_ty | Resource.close_ty] r
```

Comment:
Or, you could remove this change and do the cast in `pipe`. That avoids the need for every implementation of MGR to do the cast, which is an API change.

Reply:
wow. this is a brilliant one! 
Thank you so much for this.